packages/openid4vc-holder/src/issuance/OpenId4VciHolderService.ts

```diff
@@ -0,0 +1,703 @@
+    credentialOffer: UniformCredentialOfferPayload | string,
+    opts?: { version?: OpenId4VCIVersion }
+  ) {
+    let version = opts?.version ?? OpenId4VCIVersion.VER_1_0_11
+    const claimedCredentialOfferUrl = `openid-credential-offer://?`
+    const claimedIssuanceInitiationUrl = `openid-initiate-issuance://?`
+
+    if (
+      typeof credentialOffer === 'string' &&
+      (credentialOffer.startsWith(claimedCredentialOfferUrl) ||
+        credentialOffer.startsWith(claimedIssuanceInitiationUrl))
+    ) {
+      const credentialOfferWithBaseUrl = await CredentialOfferClient.fromURI(credentialOffer)
+      credentialOffer = credentialOfferWithBaseUrl.credential_offer
+      version = credentialOfferWithBaseUrl.version
+    }
+
+    const uniformCredentialOffer = {
+      credential_offer: typeof credentialOffer === 'string' ? undefined : credentialOffer,
+      credential_offer_uri: typeof credentialOffer === 'string' ? credentialOffer : undefined,
+    }
+
+    const credentialOfferPayload = (await assertedUniformCredentialOffer(uniformCredentialOffer)).credential_offer
+    const { metadata, issuerMetadata } = await getMetadataFromCredentialOffer(credentialOfferPayload)
+
+    this.logger.info('Fetched server metadata', {
+      issuer: metadata.issuer,
+      credentialEndpoint: metadata.credential_endpoint,
+      tokenEndpoint: metadata.token_endpoint,
+    })
+
+    this.logger.debug('Full server metadata', metadata)
+
+    const offeredCredentialsWithMetadata = getOfferedCredentialsWithMetadata(
+      credentialOfferPayload,
+      issuerMetadata,
+      version
+    )
+
+    const credentialsToRequest: CredentialToRequest[] = offeredCredentialsWithMetadata.map((offeredCredential) => {
+      const { format, types, offerType } = offeredCredential
+      if (offerType === OfferedCredentialType.InlineCredentialOffer) {
+        return { offerType, types, format }
+      } else {
+        const { id, cryptographic_binding_methods_supported, cryptographic_suites_supported } =
+          offeredCredential.credentialSupported
+
+        return { id, offerType, cryptographic_binding_methods_supported, cryptographic_suites_supported, types, format }
+      }
+    })
+
+    return {
+      metadata,
+      credentialOfferPayload,
+      credentialsToRequest,
+      version,
+    }
+  }
+
+  public async resolveAuthorizationRequest(
+    resolvedCredentialOffer: ResolvedCredentialOffer,
+    authCodeFlowOptions: AuthCodeFlowOptions
+  ): Promise<ResolvedAuthorizationRequest> {
+    const { credentialOfferPayload, metadata: _metadata } = resolvedCredentialOffer
+    const { metadata } = await getMetadataFromCredentialOffer(credentialOfferPayload, _metadata)
+    const codeVerifier = randomStringForEntropy(256)
+    const codeVerifierSha256 = Hasher.hash(TypedArrayEncoder.fromString(codeVerifier), 'sha2-256')
+    const codeChallenge = TypedArrayEncoder.toBase64URL(codeVerifierSha256)
+
+    // TODO: authdetails
```

Comment:
One last thing I am unsure there is about https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-authorization-request
Basically users can request credentials with authorization_details or scopes
The mapping between scopes and credentials is out of scope.
In theory authorization details and scopes can be used in the same request,
and the authorization details param can be constructed automatically based on the requested credentials.
Since as far as I understood it all credentials can be requested via auth details it makes sense to only support auth details not scopes. But I am unsure if there are use cases for scopes. Should we just accept  both and send the provided scopes along the automatically generated authorization_details?
Or what is your opinion on this?

Reply:
> Should we just accept both and send the provided scopes along the automatically generated authorization_details?

This seems like a good solution to me. `scopes` can still be used, but we let the user define this. 

Later on, we could create some kind of mapping a user can supply and go from there, but this should suffice for now.

It would be good to comment this code quite in detail as the behavior is a bit different.

Reply:
This is one way to implement this.
The main problem at the moment are inline credential offers.
With inline credential offers, we don't have a context, we can't provide one.
However, we also may not know the scope, which I think is possible, since not every provider will create a custom mapping of scopes to credentials.
That would mean we can't request that credential at all :( 

So in my opinion just send the context as undefined in that cases for now?
Or what are your thoughts on this?

```ts
  private getAuthDetailsFromOfferedCredential(
    credentialWithMetadata: OfferedCredentialWithMetadata,
    authDetailsLocation: string | undefined,
    version: OpenId4VCIVersion
  ): AuthDetails {
    const { format, types } = credentialWithMetadata
    const type = 'openid_credential'

    if (version < OpenId4VCIVersion.VER_1_0_11) {
      const credential_type = getV8CredentialType(credentialWithMetadata, format, version)
      return { type, credential_type, format }
    }

    const locations = authDetailsLocation ? [authDetailsLocation] : undefined
    if (format === OpenIdCredentialFormatProfile.JwtVcJson || format === OpenIdCredentialFormatProfile.JwtVcJsonLd) {
      return {
        type,
        format,
        types,
        locations,
      }
    } else if (format === OpenIdCredentialFormatProfile.LdpVc) {
      let context: string | undefined = undefined

      if (credentialWithMetadata.offerType === OfferedCredentialType.InlineCredentialOffer) {
        // Inline Credential Offers come with no context
      } else {
        if ('@context' in credentialWithMetadata.credentialSupported) {
          context = credentialWithMetadata.credentialSupported['@context'] as unknown as string
        } else {
          throw new AriesFrameworkError('Could not find @context in credentialSupported.')
        }
      }

      return {
        type,
        format,
        types,
        locations,
        '@context': context,
      }
    } else {
      throw new AriesFrameworkError(`Cannot create authorization_details. Unsupported credential format ${format}.`)
    }
  }
```

Reply:
What are the implications of returning an undefined context? Would not just make more sense for that unique scenario to just error out for now? If there is a need from the users we can check it again, but it seems quite niche that it might be a waste of time to get it perfect.